### PR TITLE
Add openidPrompt property

### LINKED ISSUE #128: proxy prompt parameter for GoogleAuth.signIn
`google-signin-aware` indirectly calls [GoogleAuth.signIn](https://developers.google.com/identity/sign-in/web/reference#googleauthsigninwzxhzdk72optionswzxhzdk73) function. The function has `prompt` parameter, but `google-signin-aware` does not pass it, and there is no way to specify its value.

I'm ok implementing it, but wanted to discuss the design first:
- add `prompt` property to `AuthEngine`
- add `prompt` to `ProxyLoginAttributes` 
- add `prompt` property of type string to `google-signin-aware` element, sync its value with `AuthEngine`
- add the property to `google-signin` element

Suggestions for a better property name are welcome.

### PR DESCRIPTION
**Pull request by nodirt**
Add prompt property to <google-signin> and <google-signin-aware>
elements. Its docs:

```
  Space-delimited, case-sensitive list of strings that
  specifies whether the the user is prompted for reauthentication
  and/or consent. The defined values are:
    none: do not display authentication or consent pages.
      This value is mutually exclusive with the rest.
    login: always prompt the user for reauthentication.
    consent: always show consent screen.
    select_account: always show account selection page.
      This enables a user who has multiple accounts to select amongst
      the multiple accounts that they might have current sessions for.
  For more information, see "prompt" parameter description in
  https://openid.net/specs/openid-connect-basic-1_0.html#RequestParameters
```

Updated demo and used it to test.

Fixes #128

### CODE REVIEW

**Comment by atotic** [comment]
Nice patch, fits well with the existing code. My only issue is the parameter name: "Prompt". Without reading the docs, I'd assume propmt would modify prompt text. Instead, it is an obscure open_id option. Could we change it to openidPrompt?

**Comment by nodirt** [comment]
Rename the property.

I am not experienced with pull requests. Do you want me to squash these commits into one or you are ok merging three commits?

**Comment by atotic** [comment]
I understand, github merging pushes into pull requests feels a bit too magical.
Your request looks good. @ebidel this looks like a useful feature, and code lgtm. Can we merge it?

File: demo/index.html [hunk 1]
```diff
@@ -146,6 +155,30 @@
       gapi.auth2.getAuthInstance().signOut();
     };
 
+    aware.addEventListener('openid-prompt-changed', function (e) {
```
Comment (ebidel) [inline demo/index.html]: `function(e) {`

File: demo/index.html [hunk 2]
```diff
@@ -146,6 +155,30 @@
...
+        }
+      }
+
+      var values = [];
+      Object.keys(aware.openidPrompt).forEach(function(k) {
+        if (aware.openidPrompt[k]) {
+          values.push(k);
+        }
+      });
+      //this.promptValue = values.join(' ');
```
Comment (ebidel) [inline demo/index.html]: remove

File: google-signin-aware.html [hunk 3]
```diff
@@ -115,6 +116,40 @@
...
+      },
+
+      set openidPrompt(val) {
+        if (typeof val !== 'string') {
+          throw new Error(
+              'openidPrompt must be a string. Received ' + typeof val);
+        }
+        if (val) {
+          var values = val.split(' ');
+          values = values.map(function(v) { return v.trim(); });
```
Comment (ebidel) [inline google-signin-aware.html]: this needs to be on newlines. no one liners.

File: google-signin-aware.html [hunk 4]
```diff
@@ -115,6 +116,40 @@
...
+          values = values.filter(function(v) { return v; });
+          var validValues = {none:0, login:0, consent: 0, select_account:0};
```
Comment (ebidel) [inline google-signin-aware.html]: spaces after `:`

**Comment by ebidel** [comment]
@addyosmani 's the maintainer. He should have a look.

**Comment by nodirt** [comment]
Addressed comments

**Comment by nodirt** [comment]
Friendly ping

**Comment by atotic** [comment]
@nodirt do not worry. addy is busy, he'll get to it and most probably merge it in.

**Pull request merged**